pkg/services/user/model.go

```diff
@@ -276,8 +277,13 @@ func (u *SignedInUser) HasRole(role roletype.RoleType) bool {
 	return u.OrgRole.Includes(role)
 }
 
+// IsRealUser returns true if the user is a real user and not a service account
 func (u *SignedInUser) IsRealUser() bool {
-	return u.UserID > 0
+	// backwards compatibility
+	// checking if userId the user is a real user
+	// previously we used to check if the UserId was 0 or -1
+	// and not a service account
+	return u.UserID > 0 && u.IsServiceAccount == false
```

Comment:
With this change we need to also update HasUniqueId to return true for service account, otherwise we will never cache permissions for a service account